tests/integration_tests/conftest.py

```diff
@@ -74,8 +74,23 @@ def disable_subp_usage(request):
     pass
 
 
+SESSION: Optional[IntegrationCloud] = None
```

Comment:
I think if we're going to go with this approach, we should eventually get rid of the `session_cloud()` fixture altogether. It's not actually doing any setup anymore so it doesn't make sense as a fixture.

That said...I'm not sure I'm ready to ditch the fixture approach entirely yet. I see the advantage and why you're approaching it this way, but this moves setup from a fairly common mechanism of using fixtures to something a little more opaque using globals. That's not the worst thing in the world either, so I guess I'm on the fence about it.

Nit: Can we name this `SESSION_CLOUD` to be a bit more descriptive?
Nit: I think we can remove the `= None` here and get rid of the `Optional`.

Reply:
> we should eventually get rid of the session_cloud() fixture altogether

Perhaps, although this allows us to keep the familiarity of fixtures while controlling setup / teardown better.

> something a little more opaque using globals

I get the hesitation to use a global variable. In this case, however, it really is a global resource - and pytest provides hook functions for managing global resources. Continuing to use the fixture to access the global resource should help keep it organized.

If we really don't want a global we could try storing it in pytest's `session` object, although I don't really see the benefit over using it as a global: this would probably be more opaque than the approach in this PR

Aside: the reaper thread that I mentioned before also currently requires storing some state in conftest to be able to pass instances to a queue and trigger thread teardown. So even if we avoid it using fixtures, I think that we will still need some way to manage global state.

> Nit: Can we name this SESSION_CLOUD to be a bit more descriptive?

Agreed, and probably just `_SESSION_CLOUD` to make it obvious that it is not intended to be used outside of the module?

> Nit: I think we can remove the = None here and get rid of the Optional.

+1

Reply:
Sounds good. Either way this is an improvement, so I'm happy to take it as-is and iterate if needed.